src/chains/ethereum_1/ethTransaction.ts

```diff
@@ -554,6 +568,7 @@ export class EthereumTransaction implements Transaction {
     // generate nonce (using privateKey) if not already present
     await this.setNonceIfEmpty(bufferToHex(privateToAddress(privateKeyBuffer)))
     this._ethereumJsTx?.sign(privateKeyBuffer)
+    this._transactionId = `0x${this._ethereumJsTx.hash(true).toString('hex')}`
```

Comment:
use ensureHexPrefix()
This logic should be moved to get transactionId() - where you can check for signature and return